packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -337,16 +355,34 @@ export function calculatePositionInternal(
     }
   }
 
-  let delta = getDelta(crossAxis, position[crossAxis], overlaySize[crossSize], boundaryDimensions, containerDimensions, padding);
+  // Determine the direction the height of the overlay can grow so that we can choose how to calculate the max height
+  let heightGrowthDirection: HeightGrowthDirection = 'bottom';
+  if (placementInfo.axis === 'top') {
+    if (placementInfo.placement === 'top') {
+      heightGrowthDirection = 'top';
+    } else if (placementInfo.placement === 'bottom') {
+      heightGrowthDirection = 'bottom';
+    }
+  } else if (placementInfo.crossAxis === 'top') {
+    if (placementInfo.crossPlacement === 'top') {
+      heightGrowthDirection = 'bottom';
+    } else if (placementInfo.crossPlacement === 'bottom') {
+      heightGrowthDirection = 'top';
+    }
+  }
```

Comment:
note that a `crossAxis=top` and `crossPlacement=top` means that the overlay's top edge is aligned with the trigger's top edge, hence why the height should grow downwards.